Source/WebInspectorUI/UserInterface/Views/ElementsTabContentView.js

```diff
@@ -127,6 +134,69 @@ WI.ElementsTabContentView = class ElementsTabContentView extends WI.ContentBrows
         super.initialLayout();
 
         this.element.appendChild(WI.ReferencePage.ElementsTab.DOMTree.createLinkElement());
+        if(WI.settings.experimentalEnableAccessibilityTreeView.value)
+            this.element.appendChild(this.createAccessibilityTreeButton());
+    }
+
+    createAccessibilityTreeButton()
+    {
+        let wrapper = document.createElement("span");
+        wrapper.className = "accessibility-tree-container";
+        this._accessibilityTreeToggleButton = new WI.ToggleButtonNavigationItem("accessibility-tree-button", 
+            WI.UIString("Go to Accessibility Tree"), 
+            WI.UIString("Go to DOM Tree"), 
+            "Images/Accessibility.svg", 
+            "Images/Accessibility-fill.svg"
+        );
+        this._accessibilityTreeToggleButton.imageType = WI.ButtonNavigationItem.ImageType.IMG;
+        this._accessibilityTreeToggleButton.addEventListener(WI.ButtonNavigationItem.Event.Clicked, (event) =>{
+            this._toggleTreeView();
+        }, this);
+        wrapper.appendChild(this._accessibilityTreeToggleButton.element);
+        return wrapper;
+    }
+
+    _toggleTreeView()
+    {
+        let mainFrame = WI.networkManager.mainFrame;
+        if (!mainFrame)
+            return;
+
+        let currentView = this.contentBrowser.currentContentView;
+        if (currentView?.domTreeOutline)
+            this._lastSelectedNode = currentView.domTreeOutline.selectedDOMNode();
+
+        const showingAccessibilityTree = this.contentBrowser.currentContentView?.representedObject instanceof WI.AccessibilityTree;
+        const newRepresentedObject = showingAccessibilityTree ? mainFrame.domTree : mainFrame.accessibilityTree;
+        
+        this.contentBrowser.showContentViewForRepresentedObject(newRepresentedObject);
+        this._accessibilityTreeToggleButton.toggled = !showingAccessibilityTree;
+        let newView = this.contentBrowser.currentContentView;
+        if (this._lastSelectedNode) {
+            if (newView.selectAndRevealDOMNode) {
+                newView.selectAndRevealDOMNode(this._lastSelectedNode);
+                
+                if (!showingAccessibilityTree) { 
```

Comment:
Can we early return here if it's already showing accessibility tree? that would make the code easier to follow.